src/kOS/Suffixed/VectorRenderer.cs

```diff
@@ -74,7 +75,13 @@ public void ScopeLost()
             // (Note that if I didn't do this,
             // then as far as C# thinks, I wouldn't be orphaned because
             // UpdateHandler is holding a reference to me.)
-            SetShow(false);
+
+            // Store the information about lost scope to be checked in the KOSUpdate method
+            // because Unity complains if we try to set visual element visibility from
+            // outside of the main thread.  That isn't a problem most of the time, but the
+            // finalizer method called by the GC (see Variable.cs ~Variable method) is
+            // apparently called from another thread, or at least Unity thinks it is.
+            scopeLost = true;
```

Comment:
I just figured out why I didn't do that initially: A local VectorRenderer added to a list, for which the VectorRenderer scope is lost but the list's scope is not lost.  Because scope is only modified by variable assignment, you can iterate through the list setting `show` to `true`, re-enabling the visibility.  If we constantly check the link count, this would eliminate the current workaround.  I'd be happy to remove this once we fix collection scope tracking though.  I'll add a todo note stating as much.